volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,331 @@
+                
+                if ATTRIBUTE_TYPE_ID[attr_type] == 'STANDARD_INFORMATION':
+                    creation_time_win = self.unpack_data(mft_record, attr_data,  'unsigned long long')
+                    modified_time_win = self.unpack_data(mft_record, attr_data+8,  'unsigned long long')
+                    altered_time_win = self.unpack_data(mft_record, attr_data+16,  'unsigned long long')
+                    access_time_win = self.unpack_data(mft_record, attr_data+24,  'unsigned long long')
+                    flags = self.unpack_data(mft_record, attr_data+32, 'unsigned short')
+                    permissions = VERBOSE_STANDARD_INFO_FLAGS.get(flags, 'Unknown')
+
+
+                    mft_entry['attributes']['SI'] = {
+                         "creation_time": self.human_date(creation_time_win),
+                         "modified_time": self.human_date(modified_time_win),
+                         "updated_time": self.human_date(altered_time_win),
+                         "accessed_time":  self.human_date(access_time_win),
+                         "flags": permissions
+                         }
+
+                if ATTRIBUTE_TYPE_ID[attr_type] == 'FILE_NAME':
+                    parent_record = self.unpack_data(mft_record, attr_data,  'unsigned long long')
+                    creation_time_win = self.unpack_data(mft_record, attr_data+8,  'unsigned long long')
+                    modified_time_win = self.unpack_data(mft_record, attr_data+16,  'unsigned long long')
+                    altered_time_win = self.unpack_data(mft_record, attr_data+24,  'unsigned long long')
+                    access_time_win = self.unpack_data(mft_record, attr_data+32,  'unsigned long long')
+
+                    name_len = self.unpack_data(mft_record, attr_data+64,  'unsigned char')
+                    name_space = self.unpack_data(mft_record, attr_data+65,  'unsigned char')
+
+                    # Unicode and partially corruprted records can break us here. 
+                    file_name = mft_record[attr_data+66:attr_data+66+(2*name_len)]
+                    file_name = utility.array_to_string(file_name)
+                    #try:
+                    #  #  file_name = file_name.replace(b'\x00', b'').decode()
+                    #except:
+                    #    file_name = str(file_name.replace(b'\x00', b''))
+
+                    flags = self.unpack_data(mft_record, attr_data+56, 'unsigned short')
+                    permissions = VERBOSE_STANDARD_INFO_FLAGS.get(flags, 'Unknown')
+
+                    mft_entry['attributes']['FN'].append(
+                        {
+                            "creation_time": self.human_date(creation_time_win),
+                            "modified_time": self.human_date(modified_time_win),
+                            "updated_time": self.human_date(altered_time_win),
+                            "accessed_time":  self.human_date(access_time_win),
```

Comment:
This should be kept/turned into a python datetime, the output renderers can deal with them and the interpretation and display of dates should be done by the UI (in case of timezone shifts, or post processing etc) rather than per-plugin.